From: Daniel van Vugt <daniel.van.vugt@canonical.com>
Date: Tue, 7 Nov 2023 16:30:01 +0200
Subject: kms/crtc: Increase default deadline evasion to 2000 microseconds

This seems to be enough to fix cursor stutter on X1 Carbon 10 and
Asus Zenbook S 13.

Origin: https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/3373
Bug: https://gitlab.gnome.org/GNOME/mutter/-/issues/3146,
     https://gitlab.gnome.org/GNOME/mutter/-/issues/3612
Bug-Ubuntu: https://bugs.launchpad.net/bugs/2040977,
            https://bugs.launchpad.net/bugs/2076318
Forwarded: yes
Last-Update: 2024-08-09
---
 src/backends/native/meta-kms-crtc.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/backends/native/meta-kms-crtc.c b/src/backends/native/meta-kms-crtc.c
index e66f8ea..52e01cd 100644
--- a/src/backends/native/meta-kms-crtc.c
+++ b/src/backends/native/meta-kms-crtc.c
@@ -28,8 +28,8 @@
 #include "backends/native/meta-kms-update-private.h"
 #include "backends/native/meta-kms-utils.h"
 
-#define DEADLINE_EVASION_US 800
-#define DEADLINE_EVASION_WITH_KMS_TOPIC_US 1000
+#define DEADLINE_EVASION_US 2000
+#define DEADLINE_EVASION_WITH_KMS_TOPIC_US 2000
 
 #define MINIMUM_REFRESH_RATE 30.f
 
